addons/hr_holidays/models/resource.py

```diff
@@ -1,10 +1,47 @@
 # -*- coding: utf-8 -*-
 # Part of Odoo. See LICENSE file for full copyright and licensing details.
 
-from odoo import fields, models
+from odoo import fields, models, api, _
+from odoo.exceptions import ValidationError
 
 
 class CalendarLeaves(models.Model):
     _inherit = "resource.calendar.leaves"
 
     holiday_id = fields.Many2one("hr.leave", string='Leave Request')
+
+    @api.constrains('date_from', 'date_to', 'calendar_id')
+    def _check_compare_dates(self):
+        all_existing_leaves = self.env['resource.calendar.leaves'].search([
+            ('resource_id', '=', False),
+            ('company_id', 'in', self.mapped('company_id.id')),
```

Comment:
You don't have to call `mapped()` on m2o, it's the default behavior when accessing them :-)
You can also use the attribute `.ids` to get the list of IDs of the recordset:

```suggestion
            ('company_id', 'in', self.company_id.ids),
```